controlplane/src/core/services/SchemaGraphPruner.ts

```diff
@@ -0,0 +1,348 @@
+    const fields: Field[] = [];
+    const schemaToBeUsed = schema || this.schema;
+
+    const types = schemaToBeUsed.getTypeMap();
+
+    for (const typeName in types) {
+      const type = types[typeName];
+      if (typeName.startsWith('__')) {
+        continue;
+      }
+
+      if (!isObjectType(type) && !isInterfaceType(type) && !isInputObjectType(type)) {
+        continue;
+      }
+
+      const fieldMap = type.getFields();
+
+      for (const fieldName in fieldMap) {
+        const field = fieldMap[fieldName];
+        if (onlyDeprecated && !field.deprecationReason) {
+          continue;
+        }
+        fields.push({
+          name: field.name,
+          typeName,
+          path: `${typeName}.${field.name}`,
+          location: {
+            line: field.astNode?.name.loc?.startToken.line,
+            column: field.astNode?.name.loc?.startToken.column,
+            endLine: field.astNode?.name.loc?.endToken.line,
+            endColumn: field.astNode?.name.loc?.endToken.column,
+          },
+          isDeprecated: !!field.deprecationReason,
+        });
+      }
+    }
+
+    return fields;
+  };
+
+  fetchUnusedFields = async ({
+    subgraphId,
+    namespaceId,
+    organizationId,
+    federatedGraphs,
+    rangeInDays,
+    addedFields,
+    severityLevel,
+  }: {
+    federatedGraphs: FederatedGraphDTO[];
+    subgraphId: string;
+    namespaceId: string;
+    organizationId: string;
+    rangeInDays: number;
+    // fields that were added in the proposed schema passed to the check command
+    addedFields: SchemaDiff[];
+    severityLevel: LintSeverityLevel;
+  }): Promise<GraphPruningIssueResult[]> => {
+    const allFields = this.getAllFields({});
+    const fieldsInGracePeriod = await this.subgraphRepo.getSubgraphFieldsInGracePeriod({ subgraphId, namespaceId });
+
+    const fieldsToBeChecked = allFields.filter((field) => {
+      return !fieldsInGracePeriod.some((f) => f.path === field.path) && !addedFields.some((f) => f.path === field.path);
+    });
+
+    const graphPruningIssues: GraphPruningIssueResult[] = [];
+
+    for (const federatedGraph of federatedGraphs) {
+      const unusedFieldsWithTypeNames = await this.usageRepo.getUnusedFields({
+        fields: fieldsToBeChecked,
+        organizationId,
+        federatedGraphId: federatedGraph.id,
+        range: rangeInDays * 24,
+      });
+      const unusedFields = allFields.filter((field) =>
+        unusedFieldsWithTypeNames.some((f) => f.name === field.name && f.typeName === field.typeName),
+      );
+      for (const field of unusedFields) {
+        graphPruningIssues.push({
+          graphPruningRuleType: 'UNUSED_FIELDS',
+          severity: severityLevel === 'error' ? LintSeverity.error : LintSeverity.warn,
+          fieldPath: field.path,
+          message: `Field ${field.name} of type ${field.typeName} has not used in the past ${rangeInDays} days`,
```

Comment:
`.. has not been used in the past x days`

Reply:
Done